apiserver/params/charmhub.go

```diff
@@ -3,75 +3,50 @@
 
 package params
 
-// TODO (hml) 2020-06-17
-// Create actual params.InfoResponse and params.ErrorResponse structs for use here.
 type CharmHubCharmInfoResult struct {
 	Result InfoResponse  `json:"result"`
 	Errors ErrorResponse `json:"errors"`
 }
 
 type InfoResponse struct {
-	Type           string        `json:"type"`
-	ID             string        `json:"id"`
-	Name           string        `json:"name"`
-	Charm          CharmHubCharm `json:"charm,omitempty"`
-	ChannelMap     []ChannelMap  `json:"channel-map"`
-	DefaultRelease ChannelMap    `json:"default-release,omitempty"`
-}
-
-type ChannelMap struct {
-	Channel  Channel  `json:"channel,omitempty"`
-	Revision Revision `json:"revision,omitempty"`
+	Type        string             `json:"type"`
+	ID          string             `json:"id"`
+	Name        string             `json:"name"`
+	Description string             `json:"description"`
+	Publisher   string             `json:"publisher"`
+	Summary     string             `json:"summary"`
+	Series      []string           `json:"series"`
+	StoreURL    string             `json:"store-url"`
+	Charm       CharmHubCharm      `json:"charm,omitempty"`
+	Bundle      CharmHubBundle     `json:"bundle,omitempty"`
+	Channels    map[string]Channel `json:"channel-map"`
+	Tracks      []string           `json:"tracks"`
 }
 
 type Channel struct {
-	Name       string   `json:"name"` // track/risk
-	Platform   Platform `json:"platform"`
-	ReleasedAt string   `json:"released-at"`
-}
-
-type Platform struct {
-	Architecture string `json:"architecture"`
-	OS           string `json:"os"`
-	Series       string `json:"series"`
-}
-
-type Revision struct {
-	CreatedAt    string     `json:"created-at"`
-	ConfigYAML   string     `json:"config-yaml"`
-	Download     Download   `json:"download"`
-	MetadataYAML string     `json:"metadata-yaml"`
-	Platforms    []Platform `json:"platforms"`
-	Revision     int        `json:"revision"`
-	Version      string     `json:"version"`
-}
-
-type Download struct {
-	HashSHA265 string `json:"hash-sha-265"`
+	ReleasedAt string `json:"released-at"`
+	Track      string `json:"track"`
+	Risk       string `json:"risk"`
+	Revision   int    `json:"revision"`
 	Size       int    `json:"size"`
-	URL        string `json:"url"`
+	Version    string `json:"version"`
 }
 
 type CharmHubCharm struct {
-	Categories  []Category        `json:"categories"`
-	Description string            `json:"description"`
-	License     string            `json:"license"`
-	Media       []Media           `json:"media"`
-	Publisher   map[string]string `json:"publisher"`
-	Summary     string            `json:"summary"`
-	UsedBy      []string          `json:"used-by"` // bundles which use the charm
+	Config      map[string]CharmOption       `json:"config"`
```

Comment:
@SimonRichardson, there may be an interface hiding in CharmOption.  Not sure how to avoid this.